packages/neon/neon/lib/src/router.dart

```diff
@@ -1,41 +1,124 @@
-part of '../neon.dart';
-
 // ignore: prefer_mixin
-class AppRouter extends RouterDelegate<Account> with ChangeNotifier, PopNavigatorRouterDelegateMixin<Account> {
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:neon/neon.dart';
+import 'package:provider/provider.dart';
+
+part 'router.g.dart';
+
+class AppRouter extends GoRouter {
   AppRouter({
-    required this.navigatorKey,
-    required this.accountsBloc,
+    required final GlobalKey<NavigatorState> navigatorKey,
+    required final AccountsBloc accountsBloc,
+  }) : super(
+          refreshListenable: StreamListenable.behaviorSubject(accountsBloc.activeAccount),
+          navigatorKey: navigatorKey,
+          initialLocation: const HomeRoute().location,
+          redirect: (final context, final state) {
+            final account = accountsBloc.activeAccount.valueOrNull;
+
+            if (account == null) {
+              return const LoginRoute().location;
+            }
+
+            if (state.location == const LoginRoute().location) {
+              return const HomeRoute().location;
+            }
+
+            return null;
+          },
+          routes: $appRoutes,
+        );
+}
+
+@immutable
+class AccountSettingsRoute extends GoRouteData {
+  const AccountSettingsRoute({
+    required this.accountid,
   });
 
-  final AccountsBloc accountsBloc;
+  final String accountid;
+
+  @override
+  Widget build(final BuildContext context, final GoRouterState state) {
+    final bloc = Provider.of<AccountsBloc>(context, listen: false);
+    final account = bloc.accounts.value.find(accountid)!;
+
+    return AccountSettingsPage(
+      bloc: bloc,
+      account: account,
+    );
+  }
+}
+
+@TypedGoRoute<HomeRoute>(
+  path: '/',
+  name: 'home',
+  routes: [
+    TypedGoRoute<SettingsRoute>(
+      path: 'settings',
+      name: 'Settings',
+      routes: [
+        TypedGoRoute<NextcloudAppSettingsRoute>(
+          path: ':appid',
```

Comment:
Account has `account/:accountid` so better to also make this `app/:appid`.

Reply:
Yeah but isn't that a bit redundant?

`/settings/apps/appID`
I just added the accounts as
`/settings/hashID` is not that meaningful.

While currently not our priority the routes will also represent the url if run in the browser.

Reply:
Hm yeah if you think about it like that then it makes sense.

Reply:
Yeah also true... just leave it as is :)